File: versioned_docs/version-1.18.x/examples/py/src/buckets_remove.py
import asyncio
from reduct import Client, ReductError


async def remove_bucket():
    # Create a client with the base URL and API token
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
        # Remove the bucket with the name "bucket-to-remove"

        bucket = await client.get_bucket("bucket-to-remove")
        await bucket.remove()

        # Check that the bucket no longer exists
        try:
            await client.get_bucket("bucket-to-remove")
        except ReductError as e:
            # The bucket should not exist anymore or still be in the process of being removed
            assert e.status_code in [404, 409]


if __name__ == "__main__":
    asyncio.run(remove_bucket())

File: versioned_docs/version-1.18.x/examples/cpp/src/buckets_remove.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>
#include <thread>
#include <chrono>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

int main() {
    // Create a client with the server URL
    auto client = IClient::Build("http://127.0.0.1:8383", {
            .api_token = "my-token"
    });

    // Remove the bucket with the name "bucket-to-remove"
    auto [bucket, get_err] = client->GetBucket("bucket-to-remove");
    assert(get_err == Error::kOk);
    auto remove_err = bucket->Remove();
    assert(remove_err == Error::kOk);

    // Check that the bucket no longer exists or is in the process of being removed
    auto [_, get_err2] = client->GetBucket("bucket-to-remove");
    assert(get_err2.code == 404 || get_err2.code == 409);
}
